pkg/cluster/cluster.go

```diff
@@ -376,10 +377,15 @@ func (c *Cluster) delete() {
 		}
 	}
 
-	if c.bm != nil {
```

Comment:
this does not seem right. we expose details again.

pass in spec to the bm we created. handle all these stuff in bm. cluster should always call

if bm != nil {
    bm.cleanup()
}